pimd/pim_mroute.c

```diff
@@ -617,10 +618,21 @@ static int pim_mroute_msg(struct pim_instance *pim, const char *buf,
 		if (!ifp || !ifp->info)
 			return 0;
 
+		connected_src = pim_if_find_connected_source(ifp, ip_hdr->ip_src);
+
+		if (!connected_src) {
+			if (PIM_DEBUG_IGMP_PACKETS) {
+				pim_inet4_dump("<src?>", ip_hdr->ip_src, ip_src_str,
+					       sizeof(ip_src_str));
+				zlog_debug("Recv IGMP packet on interface: %s from a non-connected source: %s",
```

Comment:
Can we use pI4 here?  http://docs.frrouting.org/projects/dev-guide/en/latest/logging.html